iothub/device/src/DirectMethod/DirectMethodRequest.cs

```diff
@@ -128,7 +123,7 @@ public bool TryGetPayload<T>(out T payload)
 
             try
             {
-                payload = PayloadConvention.PayloadSerializer.DeserializeToType<T>(GetPayloadAsJsonString());
+                payload = PayloadConvention.GetObject<T>(Payload);
                 return true;
             }
             catch (Exception)
```

Comment:
this seems like exception swallowing. Why catch if we are not doing anything with it? At least, it should be reported via Logs, or let the exception flow and be captured later.

Reply:
It is a "Try" .NET method pattern like `int.TryParse(...)`, meaning it will not throw.

I don't see huge value in logging out the exception.